libvast/src/format/json.cpp

```diff
@@ -189,7 +226,14 @@ caf::error add(table_slice_builder& builder, const vast::json::object& xs,
                                            "slice builder");
       continue;
     }
-    auto x = caf::visit(convert{}, *i, field.type);
+    auto x = [&] {
+      switch (policy) {
```

Comment:
Can you add a `default` case with a failing assertion?

Reply:
It's an enum class and `-Wswitch` is enabled by default. Adding a default case here is an antipattern as it silences compiler warnings.